clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -1724,16 +1724,21 @@ bool isAValidPVConstraint(const ConstraintVariable *C) {
 
 // Replace CVars and ArgumentConstraints with those in [FromCV].
 void PointerVariableConstraint::brainTransplant(ConstraintVariable *FromCV,
-                                                ProgramInfo &I) {
+                                                ProgramInfo &I,
+                                                std::string &ReasonFailed) {
   PVConstraint *From = dyn_cast<PVConstraint>(FromCV);
   assert(From != nullptr);
   CAtoms CFrom = From->getCvars();
-  assert(Vars.size() == CFrom.size());
+  if(Vars.size() != CFrom.size()) {
+    ReasonFailed = "transplanting between pointers with different depths";
```

Comment:
Is it ever possible for `ReasonFailed` to be non-empty when it gets here? In `mergeDeclaration` you are checking if it was already set, since there are recursive calls.

Reply:
(I think not, and you've probably checked; just wondering.)

Reply:
I did miss the recursive call. Partly because I have no idea what `FV` is. What is `FV`? I'll give it a modified reason if necessary with a return.

Reply:
added a function pointer fail case and an internal merge fail case, both with returns.